core/src/apps/cardano/helpers/utils.py

```diff
@@ -30,5 +35,27 @@ def to_account_path(path: List[int]) -> List[int]:
     return path[: ACCOUNT_PATH_INDEX + 1]
 
 
-def to_account_number(path: List[int]) -> int:
+def to_account_number(path: List[int]) -> Optional[int]:
+    if not SCHEMA_ADDRESS.match(path) and not SCHEMA_ACCOUNT.match(path):
```

Comment:
I would just raise a ValueError rather than letting the function return anything - if the code depending on this function expects to handle nonstandard path, it can catch the exception and handle the rendering accordingly, and if not, just let it fall through

Reply:
note: this would fail for non-standard account numbers - i.e. above 100, which is not desired - we should fix this

Reply:
Fixed in https://github.com/vacuumlabs/trezor-firmware/pull/45/commits/e90984e47b55dd88790ccc043157eea2b4033adb. I've used only a `SCHEMA_ACCOUNT_LOOSE` since it coveres both usecases and the `to_account_number` doesn't care which it is.